### PR TITLE
Fixes for building on OSX

### PR DESCRIPTION
**Pull request by swederik**
Hi Dirk,

I managed to get the ar tool on OSX to build this by adding a dummy file to the containers folder. I haven't tested these changes on Linux though, so you might want to try that before merging the PR.

I found this tip in another repository:

"Some systems, reportedly OpenBSD and Mac OS X, refuse to create
libraries without any object files.  You might get an error like:

```
ar cru .libs/libgl.a
ar: no archive members specified
```

Compiling this file, and adding its object file to the library, will
prevent the library from being empty."

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Makefile
+Makefile.in
+/Makefile
+/Makefile.in
+/aclocal.m4
+/autom4te.cache/
+/config.*
+/configure
+configure.ac
```
Comment (farindk) [inline .gitignore]: configure.ac should not be ignored

File: configure.in [hunk 2]
```diff
@@ -1,434 +0,0 @@
-dnl Process this file with autoconf to produce a configure script.
```
Comment (farindk) [inline configure.in:1]: configure.in should not be deleted (maybe renamed to configure.ac)

File: examples/x11test.cc [hunk 3]
```diff
@@ -52,7 +52,7 @@ int main(int argc,char** argv)
 	}
 
       char c;
-      if (c=win.CheckForKeypress())
+      if (c == win.CheckForKeypress())
```
Comment (farindk) [inline examples/x11test.cc]: actually, '=' is correct

**Comment by farindk** [comment]
Hi Erik,

thanks for the patch. I've commented a few things that I think are not correct. Please check, change the PR, so that I can merge it.

Dirk

File: examples/x11test.cc (hunk 3 above)
Comment (swederik) [inline examples/x11test.cc]: Ah sorry then I guess it was just clang complaining.

**Pull request merged**